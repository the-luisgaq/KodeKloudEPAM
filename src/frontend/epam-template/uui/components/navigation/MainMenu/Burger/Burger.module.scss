@use  '../../../../assets/styles/effects' as *;

.root {
    --uui-burger_menu-bg: var(--uui-neutral-90);
    --uui-burger_menu-bg-hover: var(--uui-neutral-90);
    --uui-burger_menu-overlay: var(--uui-overlay);
    --uui-burger_menu-icon: var(--uui-neutral-5);
    --uui-burger_menu-item: var(--uui-text-primary);
    //
    position: relative;
    display: flex;

    &:global(.uui-burger-menu-open) {
        z-index: 1001;
    }

    :global(.uui-burger-button) {
        width: 60px;
        height: 60px;
        display: flex;
        justify-content: center;
        cursor: pointer;
        fill: var(--uui-burger_menu-icon);
        background: transparent;
        padding: 0;
        border: 0 none;

        &:hover {
            background-color: var(--uui-burger_menu-bg-hover);
        }

        &:active {
            background-color: var(--uui-burger_menu-bg-hover);
        }

        &:focus-visible {
            @include focus-visible-effect(-2px);
        }

        :global(.uui-icon) svg {
            width: 24px;
            height: 24px;
        }
    }

    :global(.uui-burger-logo-container) {
        position: absolute;
        left: 60px;
        top: 0;
    }

    :global(.uui-burger-logo) {
        display: block;
        height: 60px;
    }
}

.burger-content {
    &:global(.uui-burger-overlay) {
        &:global(.uui-burger-overlay-visible) {
            position: fixed;
            z-index: 1000;
            top: 60px;
            bottom: 0;
            left: 0;
            right: 0;
            background: var(--uui-burger_menu-overlay);
        }
    }

    :global(.uui-burger-items) {
        position: relative;
        display: flex;
        flex-direction: column;
        top: 0;
        width: 300px;
        background: var(--uui-burger_menu-item);
        transform: translateX(-100px);

        &:global(.uui-burger-items-visible) {
            transform: translateX(0);
        }
    }
}
