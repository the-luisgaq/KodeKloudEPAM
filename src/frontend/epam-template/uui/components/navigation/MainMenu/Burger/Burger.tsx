import { withMods } from '@epam/uui-core';
import cx from 'classnames';
import { Burger as uuiBurger, BurgerProps } from '@epam/uui-components';
import css from './Burger.module.scss';
import { ReactComponent as BurgerIcon } from '@epam/assets/icons/navigation-open_side_menu-outline.svg';
import { ReactComponent as CrossIcon } from '@epam/assets/icons/navigation-close-outline.svg';

function applyBurgerMods() {
    return [cx(css.root, 'uui-burger')];
}

export const Burger = withMods<BurgerProps, BurgerProps>(uuiBurger, applyBurgerMods, () => ({
    burgerIcon: BurgerIcon,
    crossIcon: CrossIcon,
    burgerContentCx: css.burgerContent,
}));
