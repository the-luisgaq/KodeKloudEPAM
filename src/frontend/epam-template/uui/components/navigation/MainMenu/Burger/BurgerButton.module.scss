@use  '../../../../assets/styles/index' as *;

.root {
    --uui-burger_btn_caption: var(--uui-neutral-5);
    --uui-burger_btn_caption-hover: var(--uui-primary-50);
    --uui-burger_btn_caption-active: var(--uui-primary-50);
    --uui-burger_btn-border: var(--uui-neutral-30);

    --uui-burger_btn-font: var(--uui-font);
    --uui-burger_btn-font-weight: 400;

    --uui-burger_btn-icon-size: 18px;
    //
    padding: 0 24px;
    background: none;
    border: 0;
    outline: none;
    font-family: var(--uui-burger_btn-font);
    font-weight: var(--uui-burger_btn-font-weight);
    line-height: 48px;
    flex-shrink: 0;

    &:global(.uui-enabled) {
        color: var(--uui-burger_btn_caption);
        fill: var(--uui-burger_btn_caption);

        &:global(.-clickable) {
            &:hover {
                color: var(--uui-burger_btn_caption-hover);
                fill: var(--uui-burger_btn_caption-hover);
            }

            &:focus-visible {
                @include focus-visible-effect(-2px);
            }
        }
    }

    &:global(.uui-active) {
        color: var(--uui-burger_btn_caption-active);
        fill: var(--uui-burger_btn_caption-active);
    }

    :global(.uui-icon) {
        padding: 0 12px;
    }

    &.button-primary {
        font-size: 18px;
    }

    &.button-secondary {
        font-size: 14px;
        padding-left: 18px;
        margin-left: 48px;
        border-left: 1px var(--uui-burger_btn-border) solid;
    }

    &.hasIcon,
    &.dropdown {
        padding-left: 0;
    }

    &.hasIcon {
        :global(.uui-icon) {
            padding: 0 12px 0 20px;

            svg {
                height: var(--uui-burger_btn-icon-size);
                width: inherit;
            }
        }
    }
}
