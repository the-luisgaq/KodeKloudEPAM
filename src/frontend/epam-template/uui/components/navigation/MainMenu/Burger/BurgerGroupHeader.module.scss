.root {
    --uui-burger_group-text: var(--uui-text-disabled);
    --uui-burger_group-line: var(--uui-text-secondary);

    --uui-burger_group-font: var(--uui-font);
    --uui-burger_group-font-weight: 400;
}

.group-header {
    padding: 17px 0 0 0;
}

.group-name {
    font-family: var(--uui-burger_group-font);
    font-weight: var(--uui-burger_group-font-weight);
    font-size: 12px;
    color: var(--uui-burger_group-text);
    background: var(--uui-burger_menu-item);
    padding: 0 14px 0 18px;
}

.line {
    position: absolute;
    width: 100%;
    height: 1px;
    border: 0;
    background-color: var(--uui-burger_group-line);
    z-index: -1;
    margin-top: 11px;
}
