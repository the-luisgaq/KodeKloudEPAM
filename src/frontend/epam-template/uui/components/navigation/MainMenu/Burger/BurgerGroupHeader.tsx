import * as React from 'react';
import cx from 'classnames';
import css from './BurgerGroupHeader.module.scss';

interface BurgerGroupHeaderProps {
    /*
    * Defines component caption.
    */
    caption: string;
}

export function BurgerGroupHeader(props: BurgerGroupHeaderProps) {
    return (
        <div className={ cx(css.root, css.groupHeader, 'uui-burger-group-header') }>
            <hr className={ css.line } />
            <span className={ css.groupName }>{props.caption}</span>
        </div>
    );
}
