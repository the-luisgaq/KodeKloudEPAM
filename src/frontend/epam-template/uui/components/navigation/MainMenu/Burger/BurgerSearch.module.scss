.search-input {
    --uui-burger_search: var(--uui-neutral-5);
    --uui-burger_search-fill: var(--uui-text-disabled);
    --uui-burger_search-bg: var(--uui-neutral-80);
    --uui-burger_search-placeholder: var(--uui-text-disabled);

    --uui-burger_search-font: var(--uui-font);
    --uui-burger_search-font-weight: 400;
    //
    height: 60px;
    background-color: var(--uui-burger_search-bg);
    fill: var(--uui-burger_search-fill);
    padding: 0 24px;
    flex-shrink: 0;
    width: auto;

    :global(.uui-input) {
        color: var(--uui-burger_search);
        font-family: var(--uui-burger_search-font);
        font-weight: var(--uui-burger_search-font-weight);

        &::placeholder {
            color: var(--uui-burger_search-placeholder);
        }
    }

    :global(.uui-caption),
    :global(.uui-input) {
        padding: 18px 16px;
        line-height: 24px;
        font-size: 14px;
    }

    :global(.uui-icon) {
        svg {
            height: 16px;
            max-width: 16px;
        }
    }

    :global(.uui-icon-cancel) {
        svg {
            height: 24px;
            max-width: 24px;
        }

        &:hover {
            cursor: pointer;
        }
    }
}
