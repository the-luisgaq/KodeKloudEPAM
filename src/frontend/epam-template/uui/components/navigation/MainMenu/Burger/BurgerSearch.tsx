import * as React from 'react';
import { TextInput, TextInputProps } from '@epam/uui-components';
import { ReactComponent as LensIcon } from '@epam/assets/icons/action-search-outline.svg';
import { ReactComponent as CancelIcon } from '@epam/assets/icons/navigation-close-outline.svg';
import css from './BurgerSearch.module.scss';
import cx from 'classnames';

interface BurgerSearchProps extends TextInputProps {}

export function BurgerSearch(props: BurgerSearchProps) {
    return (
        <TextInput
            cx={ cx(css.searchInput, 'uui-main_menu-burger-search') }
            iconPosition="left"
            icon={ LensIcon }
            placeholder={ props.placeholder }
            value={ props.value }
            onValueChange={ props.onValueChange }
            onCancel={ props.onCancel }
            cancelIcon={ props.value && CancelIcon }
        />
    );
}
