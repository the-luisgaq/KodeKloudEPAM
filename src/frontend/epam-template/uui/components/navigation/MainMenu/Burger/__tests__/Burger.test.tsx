import React from 'react';
import { render, mockReactPortalsForSnapshots } from '@epam/uui-test-utils';
import { Burger } from '../Burger';
import { BurgerButton } from '../BurgerButton';

describe('Burger', () => {
    it('should render with maximum props', () => {
        const { mockClear } = mockReactPortalsForSnapshots();
        const { asFragment } = render(<Burger width={ 160 } renderBurgerContent={ () => <BurgerButton caption="Home" link={ { pathname: '/' } } /> } />);
        mockClear();
        expect(asFragment()).toMatchSnapshot();
    });
});
