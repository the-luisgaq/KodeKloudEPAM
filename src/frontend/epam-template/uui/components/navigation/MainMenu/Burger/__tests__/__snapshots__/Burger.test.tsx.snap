// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Burger should render with maximum props 1`] = `
<DocumentFragment>
  <div
    class="root uui-burger uui-burger-menu container"
  >
    <button
      class="uui-burger-button"
    >
      <div
        class="container uui-icon uui-enabled"
      >
        <svg
          class=""
        />
      </div>
    </button>
  </div>
</DocumentFragment>
`;
